### DISCUSSION TITLE
Error while launching on freebsd

### DISCUSSION THREAD

**Discussion by ghost** [ChristopherHX/github-act-runner#65] [Q&A] [answered]
I seem to be having some issues launching runners I have registered into GitHub : 

```
Fatal: It seems this runner was removed from GitHub, Failed to recreate Session for golang_9259d899-652e-434b-bfe2-91556ad00fa3 ( https://github.com/InfRandomness/portfolio ): Failed to Authorize, service responded with code 400: {"error":"invalid_client","error_description":"The token expired on 6/13/2022 6:29:57 PM. Current server time is 6/13/2022 8:25:51 PM."}
```

I was wondering if that occurred to anyone else but me ?

**Comment by ChristopherHX** [comment, marked as answer]
I got this kind of errors on a plan9 qemu vm with a misconfigured clock / timezone.
> The token expired on 6/13/2022 6:29:57 PM. Current server time is 6/13/2022 8:25:51 PM.

Based on the error message your UTC time from freebsd is 2 hours behind UTC. I don't know how to configure this on freebsd. Local timezones should work fine, e.g. my devices using UTC+2 and it still works.

**Reply by ghost** [reply]
my UTC time on the machine seems to be correct, I compared a correct one and the machine's one (unless I'm confused)

**Reply by ChristopherHX** [reply]
I mean the time returned by golang `time.Now().UTC()` is two hours behind UTC.
https://github.com/ChristopherHX/github-act-runner/blob/516c04ae84cd895345ad995f91a6ffcbd8fc6078/protocol/TaskAgent.go#L56

You probably will be able to authorize if you change the code to that
```go
 now := time.Now().UTC().Add(-30 * time.Second + 2 * time.Hour) 
```
_This is a Workaround and no solution_

**Reply by ghost** [reply]
> I mean the time returned by golang `time.Now().UTC()` is two hours behind UTC.
> 
> https://github.com/ChristopherHX/github-act-runner/blob/516c04ae84cd895345ad995f91a6ffcbd8fc6078/protocol/TaskAgent.go#L56
> 
> You probably will be able to authorize if you change the code to that
> 
> ```go
>  now := time.Now().UTC().Add(-30 * time.Second + 2 * time.Hour) 
> ```
> 
> _This is a Workaround and no solution_

is that a golang bug?

**Reply by ChristopherHX** [reply]
> is that a golang bug?

I have not enough information to say if is a golang bug or something else. Both freebsd on azure and locally in a Hyper-V VM doesn't caused any errors for me, however I never changed the timezone or configured anything else.
